scripts/tasks/version-check.js

```diff
@@ -6,9 +6,32 @@
  */
 
 'use strict';
+const execSync = require('child_process').execSync;
+
+function getLocalYarnVersion() {
+  return execSync('yarn --version', {
```

Comment:
This will fail if the user doesn't have yarn, right? Can we make it fail gracefully with a nice message?

We also need to make sure it works on Windows. Also some people have `yarn` bound to some Java tool, and use `yarnpkg` alias instead. Maybe that would be safer?

Reply:
Thank @gaearon for review. 

> This will fail if the user doesn't have yarn

Agreed, we can catch the error and make it fail gracefully, I will try it.

> We also need to make sure it works on Windows.

I think `yarn --version` command will support Windows.

> use yarnpkg alias instead

If user use other name instead, it's hard to trace. 🤔

Reply:
yarn installs yarnpkg and yarn by default, so you can change that to `yarnpkg --version` and checking if it's installed first (maybe `which yarnpkg` or something if that's supported on windows)

Reply:
Thank @Haroenv for reminding. yarn [v0.16.0](https://github.com/yarnpkg/yarn/blob/v0.16.0/package.json#L79-L80) started to support `yarnpkg`. It seems ok to check `yarnpkg` first.